src/client/slices/globalSlice.js

```diff
@@ -575,8 +575,33 @@ export const globalSlice = createSlice({
      * @param {number} action.payload - new mode
      */
     toggleMode: (state, action) => {
-      if (action.payload === state.mode) state.mode = IDLE;
-      else state.mode = action.payload;
+      if (action.payload === state.mode) {
```

Comment:
Please add some comments to tell others what the code here is doing.
Ex. Switch the mode back to `IDLE` and reset the current status.